packages/deployer/src/compiler.ts

```diff
@@ -162,49 +171,69 @@ export class Compiler {
         }
         const solcInstance = solc.setupMethods(requireFromString(solcjs, compilerBinFilename));
 
-        logUtils.log(`Compiling ${fileName} with Solidity v${solcVersion}...`);
-        const source = this._contractSources[fileName];
-        const input = {
-            [fileName]: source,
-        };
-        const sourcesToCompile = {
-            sources: input,
+        logUtils.log(`Compiling ${contractName} with Solidity v${solcVersion}...`);
+        const source = this._contractSources[contractName].source;
+        const absoluteFilePath = this._contractSources[contractName].absoluteFilePath;
+        const standardInput: solc.StandardInput = {
+            language: 'Solidity',
+            sources: {
+                [absoluteFilePath]: {
+                    urls: [`file://${absoluteFilePath}`],
```

Comment:
Why the `file://` prefix? Also, this isn't windows friendly. Do we want to be?

Reply:
Because if how solidity accept sources. It should be either a string with contents or the URL. So I pass the file URL and it'll read it from the FS. Making this tool windows-friendly is another PR. I'll do that when I'll create a windows VM.